apps/src/dance/ai/AiBlockPreview.tsx

```diff
@@ -1,19 +1,21 @@
 import {BlockSvg, Workspace, WorkspaceSvg} from 'blockly';
 import React, {useEffect, useRef, useState} from 'react';
 import moduleStyles from './ai-block-preview.module.scss';
+import {generateBlocksFromResult} from './utils';
 
 interface AiBlockPreviewProps {
   fadeIn: boolean;
-  generateBlocksFromResult: (workspace: Workspace) => [BlockSvg, BlockSvg];
+  resultJson: string;
   onComplete: () => void;
 }
 
 /**
  * Previews the blocks generated by the AI block in Dance Party.
  */
+// block preview rendering twice?
```

Comment:
intentional?

Reply:
Nope, fixed this. Part of the reason I pulled out `generateBlocksFromResult`

Reply:
Oh great.  I've seen occasional double sets of blocks; this is now fixed?

Reply:
I was seeing it consistently with a change I made, which I fixed. I'm not sure that it's fixed in all cases (any recollection on when you saw it)? It does seem like we might want to wipe any existing blocks from the workspace here before rendering new ones

Reply:
I don't have a specific repro but will keep an eye out.  Agreed that some extra protection against double-renders might be worthwhile.